src/librustc_mir/build/expr/into.rs

```diff
@@ -156,11 +156,16 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                     }
 
                     // execute the body, branching back to the test
-                    let unit_temp = this.unit_temp.clone();
-                    let body_block_end = unpack!(this.into(&unit_temp, body_block, body));
+                    // FIXME(#30636): this should not create or request any sort of destination at
+                    // all.
```

Comment:
Also, I'm mildly worried that writing to `destination` here will yield incorrect results for the initialization analysis, once we implement that. Seems like something like:

```
let x;

```
x = while foo { ... /* try to use `x` in here */ }
```

might succeed when it shouldn't. But hmm that particular example I guess would not, since there is at least one iteration (the first one) where the body has not executed, and thus has not written into `destination`. Hmm, I guess it's ok.